src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -305,5 +405,77 @@ await _usersAndRolesDbContext.UserRoles
                         .OnSuccess(() => _userRoleService.AddGlobalRole(userId, roleId));
                 });
         }
+
+        private async Task<Either<ActionResult, Unit>> ValidateEmail(string email)
+        {
+            if (string.IsNullOrWhiteSpace(email))
+            {
+                return ValidationActionResult(InvalidEmailAddress);
+            }
+
+            return Unit.Instance;
+        }
+
+        private async Task<Either<ActionResult, Unit>> ValidateUserDoesNotExist(string email)
+        {
+            if (_usersAndRolesDbContext.Users.Any(u => u.Email.ToLower() == email.ToLower()))
```

Comment:
Does it have to be the usersAndRolesDbContext user that you are looking for here. If not you can use `UserRepository#FindByEmail`

Reply:
This was something I refactored into its own method, so I'm not 100% sure, but `InviteUser` does handle creating creating a usersAndRolesDbContext `UserInvite`. I suspect it would work still with using FindByEmail, but I think then it's potentially relying on the two different tables being in sync? 

Not sure if contentDbContext Users and usersAndRolesDbContext Users are different? I assume these, in the DB, refer to Users and AspNetUsers respectively?

Making this change would mean adding another repository to `UserManagementService`, so I'll leave it for now. Happy to make the change if you still think it's preferable.

Reply:
You are right, they are different types of Users (Users vs. AspNetUsers)

Because this is used for inviting users to the system with system roles (e.g. to invite the BAU user) I think it's correct that this needs to check `_usersAndRolesDbContext`.

By the way though, I think you could use `AnyAsync` in this async method.

Reply:
Changed the `Any` to `AnyAsync`.